src/app/base/hooks/useWebsocketAwareQuery.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from "react";
+
+import type {
+  QueryKey,
+  QueryFunction,
+  UseQueryOptions,
+} from "@tanstack/react-query";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useSelector } from "react-redux";
+
+import type { APIError } from "@/app/base/types";
+import statusSelectors from "@/app/store/status/selectors";
+
+export function useWebsocketAwareQuery<
```

Comment:
You already defined this function in `src/app/api/query/base.ts` - is this intentional?

Reply:
I shuffled functions and files around, might've made a mistake. Let me have a look again.

Reply:
Yup, somehow didn't get deleted when I renamed it and moved to a new location.